cpp/src/parquet/arrow/writer.cc

```diff
@@ -307,7 +310,8 @@ struct ColumnWriterContext {
 Status GetLeafType(const ::arrow::DataType& type, ::arrow::Type::type* leaf_type) {
   if (type.id() == ::arrow::Type::LIST || type.id() == ::arrow::Type::STRUCT) {
     if (type.num_children() != 1) {
-      return Status::Invalid("Nested column branch had multiple children");
+      return Status::Invalid("Nested column branch had multiple children: ",
```

Comment:
`DataType` is ostreamable:
```suggestion
      return Status::Invalid("Nested column branch had multiple children: ", *type);
```